test/index.spec.js

```diff
@@ -0,0 +1,34 @@
+import trimCanvas from '../src/index.js'
+
+describe('trimCanvas', () => {
+  it('should trim whitespace (and nothing else)', () => {
+    const canvas = document.createElement('canvas')
+    const ctx = canvas.getContext('2d')
+    canvas.width = 1000
+    canvas.height = 1000
+
+    ctx.fillStyle = 'purple'
+    ctx.fillRect(450, 450, 100, 100) // 100x100 purple box in center
+    ctx.fillRect(550, 550, 100, 100) // 100x100 purple box down-right of center
+    // should look like this basically:
+    //      _
+    //     |_|_
+    //       |_|
+    //
+
+    trimCanvas(canvas)
+    expect(canvas.width).toBe(200)
+    expect(canvas.height).toBe(200)
+  })
+
+  it('should trim everything if only whitespace', () => {
+    const canvas = document.createElement('canvas')
+    canvas.width = 1000
+    canvas.height = 1000
+
+    trimCanvas(canvas)
+    // shouldn't it be 0?
```

Comment:
might want to have `scanX` / `scanY` return `-1` instead of `null` to fix this, though that might cause some bugs and not actually work as well

Reply:
okkkkk so changing them to return `-1` causes Node to print some gc errors and then just crash.... I thought `getImageData(-1, -1, -1, -1)` might error hard, but didn't think it would cause Node to crash 🤯. I'm guessing `node-canvas` has some bug or something which makes it crash instead of throwing an invalid args error...

So I guess the only real solution would be to do some non-trivial refactoring to get `cropXDiff` and `cropYDiff` to be 0 instead of 1 -- because apparently `null - null === 0` in JS.... I'm guessing this might cause some type-check issues when moving to TS. Maybe we'll refactor during the move to TS instead?

Reply:
annnd `cropXDiff` and `cropYDiff` can't be 0, because that causes `getImageData`'s [`IndexSizeError`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData#Exceptions)...

So we'd have to refactor some more so that it skips the `getImageData` call if if there's a `null` (or maybe `-1` instead) return. It'd have to skip `putImageData` too and maybe `clearRect` too, so maybe it should just have it's own branched logic 😕 That would basically just be:

```javascript
if (cropTop === null) {
  canvas.width = 0
  canvas.height = 0
  return canvas
}
```

that might interfere with the branching logic in #7 also

Reply:
Yea we'd have to add a case for if one is cloning (#7) here too. Alternatively, we could change #7 to just create a full clone at the top and then modify and return it as if it were the old canvas 🤔 

In any case, not going to make changes in this PR / version line, might do so in the TS refactor which would also allow for a potential breaking change (that might not actually be breaking and may only impact a tiny amount of users).